aya/src/programs/tc.rs

```diff
@@ -237,39 +344,111 @@ impl SchedClassifier {
 }
 
 #[derive(Debug, Hash, Eq, PartialEq)]
-pub(crate) struct TcLinkId(i32, TcAttachType, u16, u32);
+pub(crate) struct NlLinkId(u32, TcAttachType, u16, u32);
 
 #[derive(Debug)]
-struct TcLink {
-    if_index: i32,
+pub(crate) struct NlLink {
+    if_index: u32,
     attach_type: TcAttachType,
     priority: u16,
     handle: u32,
 }
 
-impl Link for TcLink {
-    type Id = TcLinkId;
+impl Link for NlLink {
+    type Id = NlLinkId;
 
     fn id(&self) -> Self::Id {
-        TcLinkId(self.if_index, self.attach_type, self.priority, self.handle)
+        NlLinkId(self.if_index, self.attach_type, self.priority, self.handle)
     }
 
     fn detach(self) -> Result<(), ProgramError> {
         unsafe {
-            netlink_qdisc_detach(self.if_index, &self.attach_type, self.priority, self.handle)
+            netlink_qdisc_detach(
+                self.if_index as i32,
+                &self.attach_type,
+                self.priority,
+                self.handle,
+            )
         }
         .map_err(|io_error| TcError::NetlinkError { io_error })?;
         Ok(())
     }
 }
 
+#[derive(Debug, Hash, Eq, PartialEq)]
+pub(crate) enum TcLinkIdInner {
+    TcxLinkId(<FdLink as Link>::Id),
+    NlLinkId(<NlLink as Link>::Id),
+}
+
+#[derive(Debug)]
+pub(crate) enum TcLinkInner {
+    TcxLink(FdLink),
```

Comment:
Shouldn't this be `FdLink(FdLink)` - this is the pattern we use elsewhere and avoid us thinking that a TcxLink is somehow special.

Reply:
Hrm TCX link _is_ a bit special in that it supports the kernel's multi-prog API, I'd like to leave it if it's ok with you

Reply:
Good point with the new trait we're already saying that a `SchedClassifierLink` supports the Kernel's Mprog API